implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceMeterRegistry.java

```diff
@@ -132,12 +132,19 @@ protected void publish() {
         }
     }
 
-    private Stream<DynatraceCustomMetric> writeMeter(Meter meter) {
+    // VisibleForTesting
+    Stream<DynatraceCustomMetric> writeMeter(Meter meter) {
         final long wallTime = clock.wallTime();
         return StreamSupport.stream(meter.measure().spliterator(), false)
+                .filter(this::isPublishable)
                 .map(ms -> createCustomMetric(meter.getId(), wallTime, ms.getValue()));
     }
 
+    private boolean isPublishable(Measurement measurement) {
```

Comment:
I've noticed that this is not a first PR where we adding `!Double.isNaN(value) && !Double.isInfinite(value)` check across different MeterRegistries. Would it make sense to move this method into `MeterRegistry` itself or into some utility/helper class and utilize in all registries (Dynatrace, Atlas, Azure, Datadog, InfluxDB, CloudWatch, etc)?

2 things can be addressed by doing this:
* consistency and maintanability since this check logic present in a single place and it's easier to update if neede;
* revise existing registries to make sure all of them have this check in place rather than addressing problems one by one.

Can be addressed as a separate item.

Reply:
@OleksiiBondar Thanks for the feedback! I thought the same before but I wasn't sure all the meter registries could be handled in the same way for NaN and infinity values. I created #1123 to check and resolve it if possible.

For the verbosity of the check logic I changed it to use `Double.isFinite()` in d70cce7. Other places can be polished separately or wait for #1123.

Reply:
Great, thank you!

Reply:
It's true that some monitoring systems are OK with NaN and Infinity, so we can't handle it in `MeterRegistry`, but the helper utility is welcome.